google/cloud/spanner/value_test.cc

```diff
@@ -904,6 +912,81 @@ TEST(Value, CommitTimestamp) {
   EXPECT_FALSE(bad.ok());
 }
 
+class ValueStreamOperator
+    : public TestWithParam<std::tuple<std::string, Value, const char*>> {
+ public:
+  static std::string TestName(
+      TestParamInfo<ValueStreamOperator::ParamType> const& info) {
+    return std::get<0>(info.param);
+  }
+};
+
+TEST_P(ValueStreamOperator, ValueType) {
+  auto param = GetParam();
+  Value value_int64(42);
+  std::stringstream scalar_value_stream;
+  scalar_value_stream << std::get<1>(param);
+  EXPECT_STREQ(scalar_value_stream.str().c_str(), std::get<2>(param));
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    ValueStreamOperatorScalar, ValueStreamOperator,
+    testing::Values(
+        std::make_tuple("Bool", Value(false), "FALSE"),
+        std::make_tuple("Int64", Value(42), "42"),
+        std::make_tuple("NullInt64", MakeNullValue<std::int64_t>(), "NULL"),
+        std::make_tuple("Float64", Value(42.0), "42"),
+        std::make_tuple("InfinityFloat64",
+                        Value(std::numeric_limits<double>::infinity()),
+                        "Infinity"),
+        std::make_tuple("NegativeInfinityFloat64",
+                        Value(-std::numeric_limits<double>::infinity()),
+                        "-Infinity"),
+        std::make_tuple("NaNFloat64", Value(std::nan("NaN")), "NaN"),
+        std::make_tuple("String", Value("Seatac Astronomy"),
+                        "\"Seatac Astronomy\""),
+        std::make_tuple("Bytes", Value(Bytes("DEADBEEF")), "B\"DEADBEEF\""),
```

Comment:
Creating a `Bytes` value from an array `a[N]` is equivalent to `Bytes(&a[0], &a[N])`.  So, `Bytes("DEADBEEF")` would mean something like `Bytes({'D','E','A','D','B','E','E','F','\0'})`.  That is, there is a NUL in there, so the expected value should not match.

Use `Bytes(std::string("DEADBEEF"))` instead, at least until we can determine whether this is a bug or a feature.